verification/curator-service/ui/src/components/bulk-case-form-fields/CaseValidationError.ts

```diff
@@ -0,0 +1,33 @@
+export default class CaseValidationError {
+    private static readonly leadingErrorText = 'Case validation failed: ';
+    private static readonly errorDelimiter = '., ';
+    private static readonly fieldMessageDelimiter = ':';
+
+    public readonly formattedIssues: string[];
+
+    constructor(public readonly rowNumber: number, apiResponse: string) {
+        const sortedErrorsByField = apiResponse
+            .substr(CaseValidationError.leadingErrorText.length)
+            .split(CaseValidationError.errorDelimiter)
+            .map((str) => {
+                const parts = str.split(
+                    CaseValidationError.fieldMessageDelimiter,
+                );
+                return {
+                    field: parts[0],
+                    message: parts.slice(1).join().trim(),
+                };
+            })
+            .sort((a, b) => {
+                return a.field > b.field ? 1 : -1;
+            });
+        this.formattedIssues = sortedErrorsByField
+            .filter(
+                (a, index) =>
+                    !sortedErrorsByField
```

Comment:
I'm not quite following what this filtering is doing, could you add a comment?

Reply:
Yep, done. Added a top-level comment detailing the creation logic.